### PR TITLE
Fixes #15806 - Webpack rails linting

### PR DESCRIPTION
**Pull request by gailsteiger**
This adds eslint to the webpack processing.
The change depends on the [Webpack PR](https://github.com/theforeman/foreman/pull/3433) being merged.

### CODE REVIEW

**Comment by dLobatog** [comment]
[test]

File: lib/tasks/pkg.rake [hunk 1]
```diff
@@ -23,6 +23,9 @@ namespace :pkg do
     ref = ENV['ref'] || 'HEAD'
     version = `git show #{ref}:VERSION`.chomp.chomp('-develop')
     raise "can't find VERSION from #{ref}" if version.length == 0
-    `git archive --prefix=foreman-#{version}/ #{ref} | bzip2 -9 > pkg/foreman-#{version}.tar.bz2`
+    `git archive --prefix=foreman-#{version}/ #{ref} > pkg/foreman-#{version}.tar`
```
Comment (dLobatog) [inline lib/tasks/pkg.rake]: These changes are not necessary

Comment (gailsteiger) [inline lib/tasks/pkg.rake]: fixed

**Comment by gailsteiger** [comment]
[test]

**Comment by tbrisker** [comment]
[test] :+1: in general, but I would suggest first updating https://theforeman.org/handbook.html#Codingstandards to add a JS section specifying what coding standards we expect in any new code, as well as to say that new JS should be placed under webpack in modules etc.

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
@gailsteiger Thanks, I've been trying it out too and :+1: - We'll be using this as the config for https://houndci.com/configuration#eslint so that it runs on every PR (only on the webpack dir) and saves us a bunch of time commenting linting failures in case the PR author didn't bother to run `npm run lint`.

Ready to merge if Jenkins :+1: 's it

File: package.json [hunk 2]
```diff
@@ -5,13 +5,17 @@
...
   "devDependencies": {
-    "stats-webpack-plugin": "^0.2.1",
-    "webpack": "^1.9.11",
-    "webpack-dev-server": "^1.9.0",
-    "expose-loader": "~0.6.0",
     "babel-core": "~6.7.2",
+    "babel-eslint": "^6.1.2",
     "babel-loader": "~6.2.4",
-    "babel-preset-es2015": "~6.6.0"
+    "babel-polyfill": "^6.13.0",
```
Comment (dLobatog) [inline package.json]: What's polyfill used for in this PR?

Comment (gailsteiger) [inline package.json]: It's not relevant here. Removed.

**Comment by dLobatog** [comment]
@gailsteiger Great, please rebase removing polyfill and push again and I'll be glad to merge. Notice you should've received an invitation to join @theforeman org, you need to make your account visible https://help.github.com/articles/publicizing-or-hiding-organization-membership/ so that Jenkins tests run after your pushes.

We'll use this on every PR through https://github.com/theforeman/foreman/pull/3761 - sounds good?

**Comment by gailsteiger** [comment]
@dLobatog it's ready. thanks for the hint on getting Jenkins to run test.

**Comment by dLobatog** [comment]
Ready to merge if Jenkins tests pass

**Comment by dLobatog** [comment]
Merged as 45ac4eb420765bb8ffd8a1a71c42ef9bdb2aa02e, thanks @gailsteiger!

**Comment by dLobatog** [comment]
All good - the failing Katello tests are due to https://github.com/Katello/katello/pull/6260 which is now fixed.

**Pull request closed without merging**